### ISSUE TITLE
Creating annotation files in KITTI format.

### ISSUE THREAD

**Issue by deeptibhegde** [mpitropov/cadc_devkit#9]
I asked the following on the repository linked in the readme. Not sure if that repo is actively maintained or not. Can I get some help on here?
Thank you

https://github.com/asvath/cadcd/issues/1#issue-692880306

**Comment by mpitropov** [comment]
Hi @deeptibhegde ,

The repository you linked is not actively maintained. It was created during the development stage of our dataset and the conversion script was made to convert one sequence to kitti format. You would have to modify it to support the current folder structure and convert all scenes. I do know one person who has already updated the script and they said they would be willing to open a PR next week.

Another option is instead of converting to KITTI, to create a dataset loader for CADC Dataset. I'm currently testing a CADC loader created for OpenPCDet https://github.com/open-mmlab/OpenPCDet but we have not yet publicly released this.

**Comment by deeptibhegde** [comment]
Okay I will try the dataset loader, thanks. Please let me know when/if the new script becomes available.

**Comment by Youngsplace0913** [comment]
Hello. 
Thank you for your great research.

May I ask when you can release the dataloader for OpenPCDet?

**Comment by mpitropov** [comment]
Sorry for the delay, I have released a data loader that you can use for OpenPCDet on my personal branch.

https://github.com/mpitropov/OpenPCDet/tree/cadc_support

If there are any issues with the branch you can open a ticket on this repo. Right now there is only one difficulty level for testing. A member within our lab is working on calculating lidar occlusion levels which would allow us to separate the testing difficulty levels. Once that is ready I will be looking at getting this merged into the official OpenPCDet repo.

**Comment by Youngsplace0913** [comment]
Thank you so much!
Thank you for your quick response.
I'll use it for reference. 
When is it completely resolved regarding occlusion, so can we use it?

Thank you

**Comment by rvalienter90** [comment]
Hi, @mpitropov thanks for your great work!  Is the KITTI converter script available?  or at least for the calibration files? Thanks!

**Comment by mpitropov** [comment]
As I answered in the other issue, a student who has graduated has created a conversion script in a repo that is no longer maintained but should still work. https://github.com/mathild7/faster_rcnn_pytorch_multimodal/blob/master/tools/cadc_unpack_all_kitti.py

When I have some time I can read the script in more detail and bring it into this repository.

**Comment by rvalienter90** [comment]
Thank you very much for your reply!.

**Comment by liby3** [comment]
Thanks very much for the job. I have tried to use the OpenPCDet for detection based on CADC according to the introduction of the repository. 
There is something wrong when I directly use the pretrained model to test the point clouds in 0002 in 2019_02_07. The log displays that "Total samples for CADC dataset: 0"

Should I copy the whole CADC data into the folder /data/?
Thank you!

**Comment by liby3** [comment]
Thanks for the work. I have been tested CADC on OpenPCDet directly based on the pretrained pointpillar model. May I ask what's the detection accuracy? The 3D and bev AP are all 0.
Thank you.

**Comment by mpitropov** [comment]
@liby3 My results for Car were about 65% for 3D mAP. The pretrained model does not work on CADC due to the differences in annotation. You will have to train a model for CADC.

I'll train a model for 80 epochs and let you know the results.

**Comment by liby3** [comment]
@mpitropov Thank you very much for your reply! I have trained a model for 50 epochs. For car detection, the average bev AP results reach nearly 45%, however, the 3D AP results were extremely low. 
I'm looking forward to your results!

**Comment by deeptibhegde** [comment]
@mpitropov , is it possible to perform inference on the CADC data using models pretrained on otherh data (e.g. KITTI)? Can you suggest which config values or code I must change to do this?
